evennia/contrib/actions/README.md

```diff
@@ -0,0 +1,372 @@
+
+The ActionCharacter's "actions.done()" method may be called to have the character end their turn. This is automatically implemented in the "done" command (see below).
+
+The "actions.stop(ongoing=True, queued=False)" method stops all of the character's ongoing and/or queued actions, based on the parameters given to it.
+The "stop" command automatically implements this method (see below).
+
+Finally, the "actions.unpuppet" command stops all of the character's cancellable actions and turns off their alarm (you should make sure that non-cancellable actions never have their "invokes_tb" item set to True, because this would prevent the alarm from being turned off and lock the room in turn-based mode). You should probably call this in your character's . 
+
+ActionRoom objects have the method "actions.display(viewed_object, message, target=None, data="", default=False)", which will show everyone in the room a processed version of the message string in the arguments list according to the room's view function (if you have set it) or in a default way if no view function exists. The target should be an object, while the data should be a string or an object. If default is set to True, and the viewed object is invisible to a given viewer, the name of the viewed object will be replaced (with "Someone" if the object is a character or "Something" otherwise) in the version of the message given to that viewer. If default is setto False, and the viewed object is invisible to the viewer, the message will not be shown to the viewer at all. Thus, the display method can be used to implement some features of stealth. 
+
+ActionRoom objects also have the method "actions.handle_invalid(action, error_string)", which will address actions that:
+* Do not have a DefaultCharacter object set as their owner (the error string is "NoOwner ")
+* Do not have a DefaultRoom object set as their room (the error string is "NoRoom ")
+* Have an owner that is not inside their room (the error string is "OwnerNotInRoom ") 
+* Have a target but not a reach (the error string is "NoReach ")
+* Do not have a target, despite has_target being set (the error string is "NoTarget ")
+* Have a target that is not inside their room (the error string is "TargetNotInRoom ")
+
+The strings of all errors encountered are concatenated and passed to the room's actions.handle_invalid method, which then notifies the MudInfo channel about the errors and attempts to clean up the action. If you know what you're doing, you can overload actions.handle_invalid to process invalid actions in a different way.
+
+
+## Actions
+
+To create your own actions, you will want to import the Action class as shown above, then subclass it. The \_\_init\_\_ function of your subclass will have a call to [super](https://docs.python.org/2/library/functions.html#super) that will include some or all of the following keyword arguments:
+
```

Comment:
Wrap your python code in three backticks, followed by the string `python` to get correct syntax highlighting in your examples. See here: https://help.github.com/articles/creating-and-highlighting-code-blocks/